gammapy/estimators/map/core.py

```diff
@@ -330,6 +331,17 @@ def reference_model(self):
         """Reference model as a `~gammapy.modeling.models.SkyModel`."""
         return self._reference_model
 
+    @reference_model.setter
```

Comment:
I would not include a setter for the reference model, as this is can lead to unexpected results if poorly handled.

Reply:
Yes, I agree. The reference model should read only. Changing the reference model requires adapting the flux norm values. 

However, we could think about an API, where the flux maps can be transferred to a new reference model, such as `FluxMaps.to_reference_model()` to similar..